lib/jekyll/commands/doctor.rb

```diff
@@ -44,6 +45,22 @@ def healthy?(site)
           ].all?
         end
 
+        def outdated_dependency_check(site)
+          return true unless testable?(site)
+
+          oudated_dep_regex = %r!\A\s*\*\s*!
+
+          Dir.chdir(site.source) do
+            exe = Gem.bin_path("bundler", "bundle")
+            _process, output = Jekyll::Utils::Exec.run("ruby", exe, "outdated")
+            output.to_s.each_line do |line|
+              next unless line =~ oudated_dep_regex
+              Jekyll.logger.warn "Outdated:", line.sub(oudated_dep_regex, "")
```

Comment:
`Outdated plugin:`, perhaps?

Reply:
The patch is not testing just *plugins* per se.. it tests every *gem* the current site **depends** on..
*plugins / theme-gem / dependencies-of-jekyll / dependencies-of-plugins-or-theme-gem*

Reply:
Hm. I’m not sure that’s useful here. I think just plugins is better, since Jekyll shouldn’t care about rake or simplecov or any other dependency. Filtering with `site.theme` and `site.plugins` would be preferable I think because it’s more relevant to Jekyll’s interests.

Reply:
By dependencies.., I was referring to the [runtime dependencies](https://github.com/jekyll/jekyll/blob/a3351cff4f7756251f7670de6cb781c99b25fde4/jekyll.gemspec#L33-L45).. *stuff that matter when our users run `bundle exec jekyll --`*..